lib/db/mysql.js

```diff
@@ -271,6 +271,23 @@ module.exports = function (log, error) {
     )
   }
 
+  var UPSERT_DEVICE = 'CALL upsertDevice_1(?, ?, ?, ?, ?, ?, ?)'
+
+  MySql.prototype.upsertDevice = function (uid, deviceId, deviceInfo) {
+    return this.write(
+      UPSERT_DEVICE,
+      [
+        uid,
+        deviceId,
+        deviceInfo.sessionTokenId,
+        deviceInfo.name,
+        deviceInfo.type,
+        deviceInfo.createdAt,
+        deviceInfo.callbackURL
```

Comment:
IUC passing `null` or `undefined` in here will cause the `COALESCE` to leave the existing value unchanged.  How would a client explicitly delete e.g. their callbackURL?  Pass in the empty string?